torch/csrc/jit/script/compiler.cpp

```diff
@@ -272,8 +272,25 @@ std::shared_ptr<SugaredValue> emitBuiltinCall(
 
   NodeKind kind(Symbol::aten(name)); // TODO: this is a guess; could it be jit?
   auto graph = method.graph();
-  auto n = graph->insertNode(graph->create(kind, inputs, 0))
-                ->setSourceLocation(std::make_shared<SourceRange>(loc));
+
+  auto source_location = std::make_shared<SourceRange>(loc);
+  std::vector<Value*> real_inputs;
+
+  // Special handling for view: since it's varargs, if we see an invocation
+  // with more than one Value* input, we emit a cat node to combine them
+  // into one input and pass that in.
+  if (kind == aten::view && inputs.size() > 2) {
```

Comment:
@That's not a good way to handle this. `view` is not the only function that has this behavior (there's also e.g. `expand`). It's a special case in our Python arg parser, that allows passing in ints as varargs if a function has a single `IntList` argument. We might want to keep that as part of the schema, or auto-generate a list of functions that trigger this case.

Reply:
I synced up with @zdevito about this offline, this code is going to get blasted away anyway